src_c/rect.c

```diff
@@ -695,6 +695,141 @@ pg_rect_collidelistall(pgRectObject *self, PyObject *args)
     return ret;
 }
 
+static SDL_Rect *
+pgRect_FromObjectAndKeyFunc(PyObject *obj, PyObject *keyfunc)
+{
+    SDL_Rect temp;
+    if (keyfunc == NULL) {
+        return pgRect_FromObject(obj, &temp);
+    }
+
+    PyObject *obj_with_rect = PyObject_CallFunctionObjArgs(keyfunc, obj, NULL);
+    if (obj_with_rect == NULL) {
+        return NULL;
```

Comment:
This branch is good, `PyObject_CallFunctionObjArgs` would have set python error already